Library/ViewModels/WithShippingRewardsCollectionViewModel.swift

```diff
@@ -429,5 +438,55 @@ private func backingAndShippingTotal(for project: Project, and reward: Reward) -
 private func allowableSortedProjectRewards(from project: Project) -> [Reward] {
   let availableRewards = project.rewards.filter { rewardIsAvailable($0) }
   let unAvailableRewards = project.rewards.filter { !rewardIsAvailable($0) }
+
   return availableRewards + unAvailableRewards
 }
+
+private func filteredRewardsByLocation(
+  _ rewards: [Reward],
+  shippingRule: ShippingRule?
+) -> [Reward] {
+  return rewards.filter { reward in
+    var shouldDisplayReward = false
+
+    let isRewardLocalOrDigital = isRewardDigital(reward) || isRewardLocalPickup(reward)
+    let isUnrestrictedShippingReward = reward.isUnRestrictedShippingPreference
+    let isRestrictedShippingReward = reward.isRestrictedShippingPreference
+
+    // return all rewards that are digital or ship anywhere in the world.
+    if isRewardLocalOrDigital || isUnrestrictedShippingReward {
+      shouldDisplayReward = true
+
+      // if add on is local pickup, ensure locations are equal.
+      if isRewardLocalPickup(reward) {
+        if let rewardLocation = reward.localPickup?.country,
+           let shippingRuleLocation = shippingRule?.location.country, rewardLocation == shippingRuleLocation {
+          shouldDisplayReward = true
+        } else {
+          shouldDisplayReward = false
+        }
+      }
+      // If restricted shipping, compare against selected shipping location.
+    } else if isRestrictedShippingReward {
+      shouldDisplayReward = rewardShipsTo(selectedLocation: shippingRule?.location.id, reward)
+    }
+
+    return shouldDisplayReward
+  }
+}
```

Comment:
This logic is mostly from the existing AddOns screen filtering logic with minor adjustments. 

If there are any suggestions for simplifying or making this easier to understand, please suggest them!